Lib/typing.py

```diff
@@ -1460,13 +1794,31 @@ def lookup_name(mapping: Mapping[KT, VT], key: KT, default: VT) -> VT:
     def __class_getitem__(cls, params):
         if not isinstance(params, tuple):
             params = (params,)
-        if not params and cls is not Tuple:
+
+        try:
+            num_class_params = len(cls.__parameters__)
+        except AttributeError:
```

Comment:
Is it possible to avoid this `try-except` block using `parameters = getattr(cls, "__parameters__", None); ...`?

Reply:
Ah, nice! Done.